File: src/dataset/cpp/collatz-conjecture_exercism-chrisnp/claude/ClaudeSonnet4_collatz_conjecture_v4.cpp
#include "collatz_conjecture.h"
#include <stdexcept>

unsigned int collatz_conjecture::steps (const int &_num) {
    if (_num <= 0) 
        throw std::domain_error("not a positive integer");
    
    unsigned int num = static_cast<unsigned int>(_num);
    unsigned int steps = 0;
    
    while (num > 1) {
        if (num & 1) {
            num = num + (num << 1) + 1;
        } else {
            num >>= 1;
        }
        ++steps;
    }
    return steps;
}

File: src/dataset/javascript/collatz-conjecture_exercism-ThomasZumsteg/claude/ClaudeSonnet4_collatz-conjecture_v4.js
class CollatzConjecture {
    steps(num) {
        if (num <= 0) {
            throw new Error('Only positive numbers are allowed');
        }
        
        let steps = 0;
        while (num !== 1) {
            num = (num & 1) ? num * 3 + 1 : num >>> 1;
            steps++;
        }
        return steps;
    }
}

module.exports = CollatzConjecture;